broadinstitute/single_cell_portal_core
# Add study publications to SCP API (SCP-5595)

This integrates studies' journal publications into SCP API, enabling better programmatic accessibility.

### Overview
[Previously](#1379), study editors and owners could add their publication's structured data, and it conveniently displayed at right in the Summary tab of the Study Overview page.  However, the data was only accessible via the UI, and not in a RESTful format.

Now publication data is available via two sets of [SCP API](https://singlecell.broadinstitute.org/single_cell/api/v1) endpoints: `/site/studies/{accession}` and a new group of CRUD endpoints anchored around `/studies/{accession}/publications`.  Implementation reuses patterns from "external resources".  These changes help decouple the frontend and backend, and enable more flexible ways to process publication data.  That promotes FAIR data.

### Limitations
There's a lot that could be improved about publications on SCP!  Some studies link to their publications in the study description, some do via "external resources"; such cases could likely be inferred and better structured in a publication object.  The sidebar UI display for publications could be refined.  Capturing DOI specifically might be nice.  "Citation" values could probably usually be automatically populated.  

This PR only intends to make existing structured data for publications accessible by API.

### Test
Automated tests verify expected behavior.  If you'd like to manually confirm:
1.  Go to a study, e.g. your local "Cellular and transcriptional diversity over the course of human lactation"
2.  Settings -> Authors/Publications
3. Add structured data for publication, e.g.:
- Title: Cellular and transcriptional diversity over the course of human lactation
- Journal: PNAS
- Citation: https://www.pnas.org/doi/10.1073/pnas.2121720119
- URL: https://doi.org/10.1073/pnas.2121720119
4. Click "Update settings"
5. Note your study's accession
6. Go to https://localhost:3000/single_cell/api/v1/site/studies/(your-study-accession)
7. Confirm a `publications` field exists in the response with the data you supplied

This satisfies SCP-5595, a Developer's Choice Days project.
